Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -205,7 +205,43 @@ public class PythonCompiler : SinglePassCompiler {
     }
 
     protected override ConcreteSyntaxTree CreateIterator(IteratorDecl iter, ConcreteSyntaxTree wr) {
-      throw new UnsupportedFeatureException(Token.NoToken, Feature.Iterators);
+      var cw = CreateClass(IdProtect(iter.EnclosingModuleDefinition.CompileName), IdName(iter), false,
+        IdName(iter), iter.TypeArgs, iter, null, iter.tok, wr) as ClassWriter;
+      Debug.Assert(cw != null, nameof(cw) + " != null");
+      var constructorWriter = cw.ConstructorWriter;
+      var w = cw.MethodWriter;
+      // here come the fields
+      Constructor ct = null;
+      foreach (var member in iter.Members) {
+        switch (member) {
+          case Field { IsGhost: false } f:
+            DeclareField(IdName(f), false, false, f.Type, f.tok, PlaceboValue(f.Type, constructorWriter, f.tok, true), constructorWriter);
+            break;
+          case Constructor constructor:
```

Comment:
I think Rustan has a PR for ghost constructors, let's make sure that it doesn't conflict with this?

Reply:
That one is only concerned with `datatype`s.